ironfish/src/wallet/wallet.ts

```diff
@@ -1492,6 +1515,24 @@ export class Wallet {
         await this.walletDb.setAccount(account, tx)
       }
 
+      if (secret) {
+        const identitySerialized = new multisig.ParticipantSecret(secret)
+          .toIdentity()
+          .serialize()
+        const multisigIdentity = await this.walletDb.getMultisigIdentity(identitySerialized)
```

Comment:
We're doing these operations (deserializing the secret, deriving the identity, reading from the db) twice in this method for the same secret. Can we combine these checks into one inside the db transaction?

Reply:
Done, I thought there might a slight downside to doing it in the transaction, but its actually better to do so. i'm passing the tx to all the getter methods which will provide a better experience incase there is another db transaction touching that secret